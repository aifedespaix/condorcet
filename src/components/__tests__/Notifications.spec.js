import { shallowMount } from "@vue/test-utils";
import Notifications from "../Notifications.vue";

describe("Notifications.vue", () => {
  it("should render when there is no notifications", () => {
    const wrapper = shallowMount(Notifications, {
      propsData: { notifications: [] }
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("should render when there is one notification", () => {
    const wrapper = shallowMount(Notifications, {
      propsData: {
        notifications: [{ status: "error", text: "Il y a eu un problème..." }]
      }
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("should render when there is more than one notification", () => {
    const wrapper = shallowMount(Notifications, {
      propsData: {
        notifications: [
          { status: "error", text: "Il y a eu un problème..." },
          { status: "success", text: "Yeaaaaah !!!" }
        ]
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
