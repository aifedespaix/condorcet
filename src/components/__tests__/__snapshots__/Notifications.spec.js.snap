// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications.vue should render when there is more than one notification 1`] = `
<div class="Notification"><span name="Transition" tag="p"><v-alert-stub value="true" type="error">
      Il y a eu un problème...
    </v-alert-stub><v-alert-stub value="true" type="success">
      Yeaaaaah !!!
    </v-alert-stub></span></div>
`;

exports[`Notifications.vue should render when there is no notifications 1`] = `<div class="Notification"><span name="Transition" tag="p"></span></div>`;

exports[`Notifications.vue should render when there is one notification 1`] = `
<div class="Notification"><span name="Transition" tag="p"><v-alert-stub value="true" type="error">
      Il y a eu un problème...
    </v-alert-stub></span></div>
`;
